src/stdune/loc.ml

```diff
@@ -94,48 +94,65 @@ let pp_file_excerpt ~context_lines ~max_lines_to_print_in_full
   let num_lines = stop.pos_lnum - start.pos_lnum in
   let whole_file = start_c = 0 && stop_c = 0 in
   if not whole_file then
-    let path = Path.of_string start.pos_fname in
-    if Path.exists path then
-      let line_num = start.pos_lnum in
-      let line_num_str = string_of_int line_num in
-      let padding_width = String.length line_num_str in
-      let line = Io.file_line path line_num in
-      if stop_c <= String.length line then
-        let len = stop_c - start_c in
-        Format.fprintf pp "%a%*s\n"
-          (pp_line padding_width) (line_num_str, line)
-          (stop_c + padding_width + 3)
-          (String.make len '^')
+    let file = start.pos_fname in
+    let res =
+      let open Result.O in
+      Result.try_with (fun () -> Sys.file_exists start.pos_fname)
+      >>= fun exists ->
+      if not exists then
+        Result.Ok ()
       else
```

Comment:
such a huge unguarded `else` makes me nervous :-) Would it be safer to wrap this clause in a `begin/end`?

Reply:
I'm actually of a fan of never allowing the ungarded if's (always using parens or begin/end) but I know some of the other maintainers aren't so keen on this style as they keep removing my parens.

In any case, there's actually a couple of these dangling elses. I added begin/end to both of them.

Reply:
I agree with avoiding unguarded ifs. Or better yet you can try to extract a local function for the `else` part, sometimes that works well. As an extra benefit that's an opportunity to invert the conditional :)

Reply:
Okay, done. I created a helper function (that's still inside the main function)